script.js

```diff
@@ -0,0 +1,149 @@
+const headerNavbar = document.getElementById('navbarSupportedContent');
+const resultBlock = document.getElementById('results');
+const pages = document.querySelector('.pagination');
+const searchButton = document.getElementById('search');
+const searchInput = document.querySelector('.form-control');
+const loader = document.querySelector('.loader');
+let currentPage = 1;
+const maxInfoLinesCount = 5;
+const API_URL = 'https://swapi.dev/api/';
+let searchMode = false;
+const LocalStorageKeys = {
+    category: 'category'
+}
+const selectedCategory = localStorage.getItem(LocalStorageKeys.category) || CATEGORIES.people;
+
+document.addEventListener("DOMContentLoaded", () => {
+    headerNavbar.querySelector(`#${selectedCategory}`).checked = true;
+    getData(selectedCategory).then((data) => {
+        renderCards(data, selectedCategory);
+        addPagination(data.count);
+    })
+});
+
+
+function getData(path) {
+    resultBlock.innerHTML = null;
+    loader.classList.remove(MODIFICATORS.hide);
+    return fetch(`${API_URL}${path}`).then((res) => res.json());
```

Comment:
add `finally` and put hide loader logic in it
add `catch` and put hide loader logic in it

Reply:
```suggestion
function getData(path) {
    loaderOn();
    return fetch(`${API_URL}${path}`).then((res) => res.json())
        .catch(() => {
            cardsContainer.innerHTML = '<span class="h3">SERVER ERROR</span>';
            loaderOff()
        })
        .finally(() => loaderOff());
}
```
Fixed